WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -199,7 +203,8 @@ def create
       redirect_to competition_path(@competition)
       return
     end
-    @registration = @competition.registrations.build(registration_params.merge(user_id: current_user.id))
+    @registration = Registration.find_by_user_id_and_competition_id(current_user.id, @competition.id) || @competition.registrations.build(user_id: current_user.id)
```

Comment:
Yes it would, so it would be like an update which I don't think is an issue. And it should never reach here unless the registration doesn't exist or is deleted.